packages/components/src/Disclosure/Disclosure.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { ReactNode, useState } from "react";
+import classnames from "classnames";
+import styles from "./Disclosure.css";
+import { Heading } from "../Heading";
+import { Icon } from "../Icon";
+
+export interface DisclosureProps {
+  /**
+   * Child content that is manged by this component.
+   */
+  readonly children: ReactNode | ReactNode[];
+
+  /**
+   * Title for the disclosure pane.
+   */
+  readonly title: string;
+
+  /**
+   * Control if the disclosure component is open or closed.
+   */
+  readonly open?: boolean;
+
+  /**
+   * This sets the default open state of the disclosure.
+   * By default the disclosure is closed.
+   * For use when the component is being used as an Uncontrolled Component.
+   * @default false
+   */
+  readonly defaultOpen?: boolean;
+
+  /**
+   * This function would toggle the open state.
+   * For use when the component is being used as a Controlled Component.
+   */
+  onRequestToggle?(isOpen: boolean): void;
+}
+
+export function Disclosure({
+  children,
+  title,
+  open,
+  defaultOpen = false,
+  onRequestToggle,
+}: DisclosureProps) {
+  const [isOpen, setOpen] = useState(open || defaultOpen);
+
+  const summaryClassName = classnames(styles.summary);
+  const detailsClassName = classnames(styles.details);
+  const contentClassName = classnames(styles.content);
+  const summaryWrapClassName = classnames(styles.summaryWrap);
```

Comment:
If we don't have any conditions on these we can just reference the `styles.foo` directly in the jsx.